src/App.vue

```diff
@@ -351,6 +362,27 @@ const { isOverDropZone } = useDropZone(dropZoneRef, {
   // whether to prevent default behavior for unhandled events
   preventDefaultForUnhandled: false,
 })
+
+// save spec to localStorage
+watch(specText, (newValue) => {
+  if (!isCleared.value && newValue && newValue.trim() !== defaultSpec.trim()) {
+    saveSpecToLocalStorage(STORAGE_KEY, newValue, () => {
+      toaster.open({
+        appearance: 'danger',
+        message: 'Failed to save specification to localStorage.',
+      })
+    })
+  }
+})
```

Comment:
issue: you'll want to add some debounce logic here. `@vueuse/core` has a `watchDebounced` you can use out of the box if that's easier